### PR TITLE
SAK-45184 Sakai 21.0 evolution

### PR DESCRIPTION
**Pull request by ern**
https://jira.sakaiproject.org/browse/SAK-45184

Will need someone to update the oracle script

### CODE REVIEW

**Comment by bjones86** [comment]
Curious, why are you removing a whole bunch of default values from all these columns?

File: docs/conversion/sakai_21_mysql_conversion.sql [hunk 1]
```diff
@@ -65,3 +65,186 @@ UPDATE rbc_tool_item_rbc_assoc SET created = NOW() WHERE created IS NULL;
...
+ALTER TABLE SSQ_QUESTION MODIFY REQUIRED BIT(1) NOT NULL;
+ALTER TABLE SSQ_QUESTION ALTER REQUIRED DROP DEFAULT;
+
+ALTER TABLE SSQ_USER_ANSWER MODIFY SITE_ID VARCHAR(255) NOT NULL;
+ALTER TABLE SSQ_USER_ANSWER ALTER SITE_ID DROP DEFAULT;
+ALTER TABLE SSQ_USER_ANSWER MODIFY USER_ID VARCHAR(255) NOT NULL;
+ALTER TABLE SSQ_USER_ANSWER ALTER USER_ID DROP DEFAULT;
+
+ALTER TABLE SSQ_SITETYPE_QUESTIONS MODIFY SITE_TYPE VARCHAR(255) NOT NULL;
+ALTER TABLE SSQ_SITETYPE_QUESTIONS ALTER SITE_TYPE DROP DEFAULT;
```
Comment (bjones86) [inline docs/conversion/sakai_21_mysql_conversion.sql:235]: Comment here indicates these are for sitestats, but sitestats tables start with SST not SSQ

File: docs/conversion/sakai_21_mysql_conversion.sql [hunk 2]
```diff
@@ -65,3 +65,186 @@ UPDATE rbc_tool_item_rbc_assoc SET created = NOW() WHERE created IS NULL;
...
+ALTER TABLE rbc_evaluation ADD status INT NOT NULL;
+UPDATE rbc_evaluation SET status = 2;
+-- END SAK-42371
+
+-- START SAK-45137
+UPDATE rbc_rating SET title = '' WHERE title IS NULL;
+ALTER TABLE rbc_rating MODIFY title varchar(255) NOT NULL;
+UPDATE rbc_rating SET points = 0 WHERE points IS NULL;
+ALTER TABLE rbc_rating MODIFY points double NOT NULL;
+-- END SAK-45137
```
Comment (bjones86) [inline docs/conversion/sakai_21_mysql_conversion.sql:249]: I've already provided the Oracle equivalent of this snippet in a different ticket for Adrian.

File: docs/conversion/sakai_21_mysql_conversion.sql (hunk 1 above)
Comment (ern) [reply docs/conversion/sakai_21_mysql_conversion.sql:232]: i thought it was sitestats! its really site setup questions!

**Comment by ern** [comment]
It's not that I am removing anything it's simply that these fields never included default in the hbm.

**Comment by davidpbauer** [comment]
I've added the Oracle conversion to this PR https://github.com/sakaiproject/sakai-reference/pull/114

**Pull request merged**